File: 01_ProgrammingBasics/_05_LoopsPartTwoExercise/_01_OldBooks.java
package _01_ProgrammingBasics._05_LoopsPartTwoExercise;

import java.util.Scanner;

public class _01_OldBooks {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        String book = scan.nextLine();
        String favoriteBook = scan.nextLine();

        // Staging
        int count = 0;

        // Use a while loop to search for the favorite book
        while (!favoriteBook.equals(book)) {
            if (favoriteBook.equals("No More Books")) {
                System.out.println("The book you search is not here!");
                System.out.printf("You checked %d books.", count);
                break;
            }
            
            count++;
            favoriteBook = scan.nextLine();
        }

        // Display the result based on whether the favorite book was found
        if (favoriteBook.equals(book)) {
            System.out.printf("You checked %d books and found it.", count);
        }

        scan.close();
    }
}

File: 23_C#Basics/_05_LoopsPartTwoExercise/_01_OldBooks/_01_OldBooks.cs
using System;

public class _01_OldBooks
{
    public static void Main(string[] args)
    {
        // Read input values
        string book = Console.ReadLine();
        string favoriteBook = Console.ReadLine();

        // Staging
        int count = 0;

        // Use a while loop to search for the favorite book
        while (!favoriteBook.Equals(book))
        {
            if (favoriteBook.Equals("No More Books"))
            {
                Console.WriteLine("The book you search is not here!");
                Console.WriteLine($"You checked {count} books.");
                break;
            }

            count++;
            favoriteBook = Console.ReadLine();
        }

        // Display the result based on whether the favorite book was found
        if (favoriteBook.Equals(book))
        {
            Console.WriteLine($"You checked {count} books and found it.");
        }
    }
}
